### DISCUSSION TITLE
GetLegalMovesNonAlloc, stackalloc and unsafe

### DISCUSSION THREAD

**Discussion by rokbok** [SebLague/Chess-Challenge#378] [Q&A] [answered]
The docs for `GetLegalMovesNonAlloc` say "...but allows you to be more efficient with memory by allocating moves on the stack rather than the heap."

Howver, using `stackalloc` would require `unsafe`, which is a rule violation? Or am I missing something?

**Comment by SebLague** [comment, marked as answer]
You can use stackalloc without unsafe context when used with Span

**Reply by rokbok** [reply]
Awesome, thank you, I did not know this
